gordo_components/model/anomaly/diff.py

```diff
@@ -0,0 +1,157 @@
+        based approach to anomaly detection.
+
+        It trains a ``scaler`` to the target **after** training, purely for
+        error calculations. The underlying ``base_estimator`` is trained
+        with the original, unscaled, ``y``.
+
+        Parameters
+        ----------
+        base_estimator: sklearn.base.BaseEstimator
+            The model to which normal ``.fit``, ``.predict`` methods will be used.
+        scaler: sklearn.base.TransformerMixn
+            Defaults to ``sklearn.preprocessing.MinMaxScaler``
+            Used for transforming model output and the original ``y`` to calculate
+            the difference/error in model output vs expected.
+        """
+        self.base_estimator = base_estimator
+        self.scaler = scaler
+
+    def __getattr__(self, item):
+        """
+        Treat this model as transparent into base_estimator unless
+        referring to something owned by this object
+        """
+        if item in self.__dict__:
+            return getattr(self, item)
+        else:
+            return getattr(self.base_estimator, item)
+
+    def get_metadata(self):
+        if isinstance(self.base_estimator, GordoBase):
+            return self.base_estimator.get_metadata()
+        else:
+            return {}  # TODO: Fill this in.
+
+    def score(
+        self,
+        X: Union[np.ndarray, pd.DataFrame],
+        y: Union[np.ndarray, pd.DataFrame],
+        sample_weight: Optional[np.ndarray] = None,
+    ) -> float:
+        return self.base_estimator.score(X, y)
+
+    def save_to_dir(self, directory: str):
+        serializer.dump(self.base_estimator, os.path.join(directory, "base_estimator"))
+        serializer.dump(self.scaler, os.path.join(directory, "scaler"))
+        with open(os.path.join(directory, "params.json"), "w") as f:
+            params = {
+                k: v
+                for k, v in self.get_params().items()
+                if k not in ("base_estimator", "scaler")
+            }
+            json.dump(params, f)
+
+    @classmethod
+    def load_from_dir(cls, directory: str):
+        with open(os.path.join(directory, "params.json"), "r") as f:
+            params = json.load(f)
+        params["base_estimator"] = serializer.load(
+            os.path.join(directory, "base_estimator")
+        )
+        params["scaler"] = serializer.load(os.path.join(directory, "scaler"))
+        return cls(**params)
+
+    def get_params(self, deep=True):
+        return {"base_estimator": self.base_estimator, "scaler": self.scaler}
+
+    def fit(self, X: np.ndarray, y: np.ndarray):
+        self.base_estimator.fit(X, y)
+        self.scaler.fit(y)  # Scaler is used for calculating errors in .anomaly()
+        return self
+
+    def anomaly(
+        self, X: pd.DataFrame, y: pd.DataFrame, frequency: Optional[timedelta] = None
+    ) -> pd.DataFrame:
+        """
+        Create an anomaly dataframe from the base provided dataframe.
```

Comment:
Why does this not do self.predict and use the model output from there? It seems clunky for me when I am in the terminal that I first have to run `.predict`, take the output from that and put it into a dataframe and then use the `anomaly` on the same object I just got some output from. 

I understand that this comes from the web-server, but maybe ditch the whole "call basic endpoint first then pass on", and rather just call the `anomaly` method directly?

Reply:
I'll buy what you're sellin'.

Reply:
Done with https://github.com/equinor/gordo-components/pull/368/commits/49d2de7465a1b61cf25420b8fedbf596fd6c5583